jhotdraw-samples/jhotdraw-samples-misc/src/main/java/org/jhotdraw/samples/pert/PertPanel.java

```diff
@@ -76,27 +76,20 @@ public PertPanel() {
     JMenu m = new JMenu(labels.getString("view.zoomFactor.text"));
     JRadioButtonMenuItem rbmi;
     ButtonGroup group = new ButtonGroup();
+
     m.add(rbmi = new JRadioButtonMenuItem(new ZoomAction(editor, 0.1, null)));
     group.add(rbmi);
-    m.add(rbmi = new JRadioButtonMenuItem(new ZoomAction(editor, 0.25, null)));
-    group.add(rbmi);
-    m.add(rbmi = new JRadioButtonMenuItem(new ZoomAction(editor, 0.5, null)));
-    group.add(rbmi);
-    m.add(rbmi = new JRadioButtonMenuItem(new ZoomAction(editor, 0.75, null)));
-    group.add(rbmi);
-    m.add(rbmi = new JRadioButtonMenuItem(new ZoomAction(editor, 1.0, null)));
-    rbmi.setSelected(true);
-    group.add(rbmi);
-    m.add(rbmi = new JRadioButtonMenuItem(new ZoomAction(editor, 1.25, null)));
-    group.add(rbmi);
-    m.add(rbmi = new JRadioButtonMenuItem(new ZoomAction(editor, 1.5, null)));
-    group.add(rbmi);
-    m.add(rbmi = new JRadioButtonMenuItem(new ZoomAction(editor, 2, null)));
-    group.add(rbmi);
-    m.add(rbmi = new JRadioButtonMenuItem(new ZoomAction(editor, 3, null)));
-    group.add(rbmi);
-    m.add(rbmi = new JRadioButtonMenuItem(new ZoomAction(editor, 4, null)));
-    group.add(rbmi);
+    for (double zoomLevel = 0.25; zoomLevel <= 4.0; zoomLevel += 0.25) {
+      if (zoomLevel == 1.0) {
+        m.add(rbmi = new JRadioButtonMenuItem(new ZoomAction(editor, zoomLevel, null)));
+        rbmi.setSelected(true);
+        group.add(rbmi);
+      } else {
+        m.add(rbmi = new JRadioButtonMenuItem(new ZoomAction(editor, zoomLevel, null)));
+        group.add(rbmi);
+      }
+    }
+
```

Comment:
Could we implement the above loop in the following way instead to remove the redundant lines of code just to make it more concise?
for (double zoomLevel = 0.25; zoomLevel <= 4.0; zoomLevel += 0.25) {
    m.add(rbmi = new JRadioButtonMenuItem(new ZoomAction(editor, zoomLevel, null)));
    group.add(rbmi);
    if (zoomLevel == 1.0) {
        rbmi.setSelected(true);
    }
}

Reply:
Yes, it is more concise. The proposed revision significantly streamlines the loop by eliminating repetitive code and directly setting the selected state for the zoomLevel of 1.0. This refactoring maintains the original functionality—adding radio button menu items for each zoom level and marking the item corresponding to a zoomLevel of 1.0 as selected—while improving code readability and maintainability. By consolidating the addition of menu items and selection logic into a single, clear block, the change enhances the efficiency of the code, making it easier to understand and modify in the future.